datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -100,18 +108,44 @@ def __init__(
         # we use the same consistent snapshot to download targets.
         self.__updater.refresh()
 
-    def __download_with_tuf(self, target_relpath):
+    def __compute_target_paths(self, target_relpath):
         # The path used to query TUF needs to be a path-relative-URL string
         # (https://url.spec.whatwg.org/#path-relative-url-string), which means the path
         # separator *must* be `/` and only `/`.
         # This is a defensive measure to make things work even if the provided `target_relpath`
         # is a platform-specific filesystem path.
         tuf_target_path = pathlib.PurePath(target_relpath).as_posix()
+        target_abspath = os.path.join(self.__targets_dir, tuf_target_path)
+
+        return tuf_target_path, target_abspath
+
+    def __download_without_tuf(self, target_relpath):
+        assert isinstance(self.__updater._target_base_url, str), self.__updater._target_base_url
+
+        tuf_target_path, target_abspath = self.__compute_target_paths(target_relpath)
+
+        # reproducing how the "self.__updater.download_target" method computes the URL
```

Comment:
Will download_target and __download_without_tuf have the same way to compute the URL for the foreseeable future? Would it make sense to have a helper function that is called by both functions to compute the URL so that this stays the case?

Reply:
`download_target` is part of TUF so we cannot really control how it resolves the URL inside. However I don't expect significant changes in the process. Plus, we control where we store the files in the S3 bucket anyway.

Also note that even now the two functions don't fetch the same URL anyway: TUF will prepend the file hash to the target name. Resolving the hash requires to run TUF (read the metadata files etc.). Downloading the file in its non-hash-prefixed version is better for reliability. We'll probably have to add a test in the part that uploads new integrations to make sure it always uploads the non-hash prefixed file which ` __download_without_tuf` relies on but not TUF.